Source/Scene/CesiumIon.js

```diff
@@ -0,0 +1,267 @@
+        Check.defined('assetId', assetId);
+
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var serverUrl = defaultValue(options.serverUrl, CesiumIon.defaultServerUrl);
+        var accessToken = defaultValue(options.accessToken, CesiumIon.defaultAccessToken);
+
+        var resourceOptions = {
+            url: serverUrl + '/v1/assets/' + assetId + '/endpoint'
+        };
+
+        if (defined(accessToken)) {
+            resourceOptions.queryParameters = { access_token: accessToken };
+        }
+
+        var endpointResource = new Resource(resourceOptions);
+        return CesiumIon._loadJson(endpointResource)
+            .then(function(endpoint) {
+                return CesiumIonResource.create(endpoint, endpointResource);
+            });
+    };
+
+    /**
+     * Creates an {@link ImageryProvider} representing a Cesium ion imagery asset.
+     * Unlike {@link CesiumIon.createResource}, this function supports external asset functionality.
+     *
+     * @param {Number} assetId The Cesium ion asset id.
+     * @param {Object} [options] An object with the following properties:
+     * @param {String} [options.accessToken=CesiumIon.defaultAccessToken] The access token to use.
+     * @param {String} [options.serverUrl=CesiumIon.defaultServerUrl] The url to the Cesium ion API server.
+     * @returns {Promise<ImageryProvider>} A promise to an imagery provider presenting the requested Cesium ion Asset.
+     *
+     * @example
+     * //Load an ImageryProvider with asset ID of 2347923
+     * Cesium.CesiumIon.createImageryProvider(2347923)
+     *   .then(function (imageryProvider) {
+     *     viewer.imageryLayers.addProvider(imageryProvider);
+     *   });
+     */
+    CesiumIon.createImageryProvider = function(assetId, options) {
+        return CesiumIon.createResource(assetId, options)
+            .then(function(resource) {
+                return resource.createImageryProvider();
+            });
+    };
+
+    /**
+     * A {@link Resource} instance that encapsulates Cesium ion asset
+     * creation and automatic refresh token handling.
+     *
+     * @private
+     */
+    function CesiumIonResource(options, endpoint, endpointResource) {
+        Resource.call(this, options);
+
+        this.ionData = {
+            endpoint: endpoint,
+            _endpointResource: endpointResource,
+            _pendingPromise: undefined,
+            _root: this
+        };
+    }
+
+    CesiumIonResource.create = function(endpoint, endpointResource) {
+        var options = {
+            url: endpoint.url,
+            retryCallback: createRetryCallback(endpoint, endpointResource),
+            retryAttempts: 1
+        };
+
+        if (defined(endpoint.accessToken)) {
+            options.queryParameters = { access_token: endpoint.accessToken };
+        }
+
+        return new CesiumIonResource(options, endpoint, endpointResource);
+    };
+
+    if (defined(Object.create)) {
```

Comment:
What doesn't have `Object.create`? Older versions of ie?
Should we have a polyfill?

Reply:
This just follows current practices already used elsewhere (agian like `RuntimeError` and `DeveloperError`, switching to a different convention is outside the scope of this PR.